xbmc/video/VideoDatabase.cpp

```diff
@@ -2347,7 +2347,14 @@ int CVideoDatabase::SetDetailsForMovie(const std::string& strFilenameAndPath, CV
       // add art if not available
       std::map<std::string, std::string> setArt;
```

Comment:
See my comment [here](https://github.com/xbmc/xbmc/pull/13564#pullrequestreview-162892988) to elide an unnecessary copy later.

Reply:
I only have about a cram session's worth of understanding on this so I'm not seeing any copies here, could you explain further?

I do see that I could add a `HasArtForItem` function that will avoid creating a map if the set already does have artwork in the database, since in this case we don't actually need the existing art.